lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -331,16 +334,43 @@ export class BrowserStorage extends CacheManager {
      * Create authorityKey to cache authority
      * @param state
      */
-    generateAuthorityKey(state: string): string {
-        return `${TemporaryCacheKeys.AUTHORITY}${Constants.RESOURCE_DELIM}${state}`;
+    generateAuthorityKey(stateString: string): string {
+        const {
+            libraryState: {
+                id: stateId
+            }
+        } = ProtocolUtils.parseRequestState(this.cryptoImpl, stateString);
+
+        return this.generateCacheKey(`${TemporaryCacheKeys.AUTHORITY}.${stateId}`);
     }
 
     /**
      * Create Nonce key to cache nonce
      * @param state
      */
-    generateNonceKey(state: string): string {
-        return `${TemporaryCacheKeys.NONCE_IDTOKEN}${Constants.RESOURCE_DELIM}${state}`;
+    generateNonceKey(stateString: string): string {
+        const {
+            libraryState: {
+                id: stateId
+            }
+        } = ProtocolUtils.parseRequestState(this.cryptoImpl, stateString);
+
+        return this.generateCacheKey(`${TemporaryCacheKeys.NONCE_IDTOKEN}.${stateId}`);
+    }
+
+    /**
+     * Creates full cache key for the request state
+     * @param stateString State string for the request
+     */
+    generateStateKey(stateString: string): string {
+        // Use the library state id to key temp storage for uniqueness for multiple concurrent requests
+        const {
+            libraryState: {
+                id: stateId
```

Comment:
Using stateId since 1) its different from teh state string itself 2) it wont have encoding issues 3) its shorter (important for cookies)